### DISCUSSION TITLE
Safari + Keyboard events = broken?

### DISCUSSION THREAD

**Discussion by huphtur** [alpinejs/alpine#3338] [7. Bugs (archived)]
Pressing ESC key on MacOS Safari 16.1 does not close the [modal example](https://alpinejs.dev/component/modal).

**Comment by magnusskare** [comment]
I think Safari requires you to explicitly focus one of the buttons to bring focus into the modal before it reacts to keydown events, as it otherwise considers the modal not in focus. So this might be more related to how `x-trap` works in Safari, rather than keyboard events themselves.

**Reply by huphtur** [reply]
I think you may be onto something, non of the [focus plugin](https://alpinejs.dev/plugins/focus) examples work on Safari ☹️

**Reply by ekwoka** [reply]
MDN recommends on Safari to do `el.click() && el.focus()` to get the focus to actually go through, so maybe that would be a worthwhile update.
